README.md
# Syssec 마작 리더보드
[작혼](https://mahjongsoul.com/) 게임 로그 분석을 통해 Syssec의 마작 최강자를 기리는 리더보드입니다 🍕

현재는 **4인 마작** 기록을 기준으로 테이블이 구성되어있고, 3인 마작 로그는 수집만 하고 있습니다.

심심할 때 가끔 만지므로 업데이트 속도는 보장할 수 없습니다!

# TODOS
- [ ] ELO 구현
- [ ] ISSUE 이용해서 로그 올릴 수 있도록 자동화
- [ ] 역만의 전당 페이지 추가
- [ ] 3인 마작 테이블 수치 계산


docs/CHANGELOG.md
## [1.6.2](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.6.1...v1.6.2) (2022-05-16)


### Bug Fixes

* **app:** change image base path ([204fb8c](https://github.com/KOREAN139/syssec-leaderboard/commit/204fb8cfdf550a615eac8e12a405284fa02208ac))

## [1.6.1](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.6.0...v1.6.1) (2022-05-16)


### Bug Fixes

* **app:** 이미지 로더 akamai로 교체 ([#37](https://github.com/KOREAN139/syssec-leaderboard/issues/37)) ([501b33e](https://github.com/KOREAN139/syssec-leaderboard/commit/501b33e279cd2740352049df20817439d75b9916))

# [1.6.0](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.5.1...v1.6.0) (2022-05-16)


### Features

* **component:** 역만 타일 이미지 렌더링 ([#36](https://github.com/KOREAN139/syssec-leaderboard/issues/36)) ([edb8cf4](https://github.com/KOREAN139/syssec-leaderboard/commit/edb8cf45538b95ea9bdbd2aff49dca00c6e30b90))

## [1.5.1](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.5.0...v1.5.1) (2022-05-12)


### Bug Fixes

* **page:** 리사이즈시 vh 세팅 대응 ([#33](https://github.com/KOREAN139/syssec-leaderboard/issues/33)) ([baf2008](https://github.com/KOREAN139/syssec-leaderboard/commit/baf200834fe6c5f0cf3f92fd4a6cea170170536a))

# [1.5.0](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.4.1...v1.5.0) (2022-05-12)


### Features

* **page:** 스크롤로 테이블 넘기는 기능 구현 ([#32](https://github.com/KOREAN139/syssec-leaderboard/issues/32)) ([f9af2ad](https://github.com/KOREAN139/syssec-leaderboard/commit/f9af2ad976a129380b4e194cd9ca0d5ed38e1315))

## [1.4.1](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.4.0...v1.4.1) (2022-05-12)


### Bug Fixes

* **page:** 랜딩페이지 패딩 제거 ([#30](https://github.com/KOREAN139/syssec-leaderboard/issues/30)) ([a2bf0aa](https://github.com/KOREAN139/syssec-leaderboard/commit/a2bf0aa333bc9178f9d4439f3475953e3ccdb180))

# [1.4.0](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.3.2...v1.4.0) (2022-05-12)


### Features

* **page:** 역만 테이블 와이어프레임 추가 ([#25](https://github.com/KOREAN139/syssec-leaderboard/issues/25)) ([a0e65f8](https://github.com/KOREAN139/syssec-leaderboard/commit/a0e65f8c136b01e867b0465143798df01180b4f0))

## [1.3.2](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.3.1...v1.3.2) (2022-05-10)


### Bug Fixes

* **analyzer:** 키 삭제 전 존재 확인하는 로직 추가 ([#20](https://github.com/KOREAN139/syssec-leaderboard/issues/20)) ([b41ce81](https://github.com/KOREAN139/syssec-leaderboard/commit/b41ce810119dbc4a00ff6cadcac3ba96c7a47d45))

## [1.3.1](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.3.0...v1.3.1) (2022-05-10)


### Bug Fixes

* **fetcher:** 1인 게임 로그는 스킵하도록 수정 ([#19](https://github.com/KOREAN139/syssec-leaderboard/issues/19)) ([24a551e](https://github.com/KOREAN139/syssec-leaderboard/commit/24a551ef33023a353d3959ad5210be9a9aaf0138))

# [1.3.0](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.2.0...v1.3.0) (2022-05-10)


### Features

* **scraper:** UUID 스크래핑 userscript 작성 ([#16](https://github.com/KOREAN139/syssec-leaderboard/issues/16)) ([fe1b137](https://github.com/KOREAN139/syssec-leaderboard/commit/fe1b1377d018f2f40637beeca1c8aa639c503e0e))

# [1.2.0](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.1.0...v1.2.0) (2022-05-05)


### Features

* **analyzer:** 역만 정보를 저장 ([#14](https://github.com/KOREAN139/syssec-leaderboard/issues/14)) ([1b89ddf](https://github.com/KOREAN139/syssec-leaderboard/commit/1b89ddfb14ac08b05e3e8e5e0e3f08bb65523fa3))

# [1.1.0](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.0.6...v1.1.0) (2022-05-05)


### Features

* **analyzer:** 역 정보를 저장합니다 ([#12](https://github.com/KOREAN139/syssec-leaderboard/issues/12)) ([6afbab0](https://github.com/KOREAN139/syssec-leaderboard/commit/6afbab048aaa42ddf365238ec602870575462342))

## [1.0.6](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.0.5...v1.0.6) (2022-05-05)


### Bug Fixes

* **fetcher:** RecordNewRound에 방장 seat 세팅 ([#11](https://github.com/KOREAN139/syssec-leaderboard/issues/11)) ([3c9fb59](https://github.com/KOREAN139/syssec-leaderboard/commit/3c9fb59816acc68d65c8bfe6e72f01525710faf5))

## [1.0.5](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.0.4...v1.0.5) (2022-05-04)


### Bug Fixes

* **util:** 순위 로직에 12% 비교로직 추가 ([#10](https://github.com/KOREAN139/syssec-leaderboard/issues/10)) ([652bb5e](https://github.com/KOREAN139/syssec-leaderboard/commit/652bb5e867cf876fdc5d117b95c7de2628845092))

## [1.0.4](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.0.3...v1.0.4) (2022-05-04)


### Bug Fixes

* **page:** 테이블 이슈 해결 ([#9](https://github.com/KOREAN139/syssec-leaderboard/issues/9)) ([d5559de](https://github.com/KOREAN139/syssec-leaderboard/commit/d5559de7f05e78b220294011278ec8e261939edb))

## [1.0.3](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.0.2...v1.0.3) (2022-05-03)


### Bug Fixes

* **build:** add asset prefix to serve static files ([2797631](https://github.com/KOREAN139/syssec-leaderboard/commit/27976314e9e315a268ad3a57429dd0f017867937))

## [1.0.2](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.0.1...v1.0.2) (2022-05-03)


### Bug Fixes

* **gh-pages:** add homepage in package.json ([7b351d9](https://github.com/KOREAN139/syssec-leaderboard/commit/7b351d9d40055e86ddedc1caa56567f393e0b429))

## [1.0.1](https://github.com/KOREAN139/syssec-leaderboard/compare/v1.0.0...v1.0.1) (2022-05-03)


### Bug Fixes

* **build:** set deploy branch check properly ([8916df3](https://github.com/KOREAN139/syssec-leaderboard/commit/8916df3febcd42314fca4ad2ce0b42b08b84c0b6))

# 1.0.0 (2022-05-03)


### Bug Fixes

* **build:** 빌드 에러 해결 ([#6](https://github.com/KOREAN139/syssec-leaderboard/issues/6)) ([52047cf](https://github.com/KOREAN139/syssec-leaderboard/commit/52047cf753067ca127359559e77c304c854e2e37))


### Features


log-analyzer/yaku.py
YAKU_MAP = {
    1: "멘젠쯔모",
    2: "리치",
    3: "창깡",
    4: "영상개화",
    5: "해저로월",
    6: "하저로어",
    7: "역패: 백",
    8: "역패: 발",
    9: "역패: 중",
    10: "자풍패",
    11: "장풍패",
    12: "탕야오",
    13: "이페코",
    14: "핑후",
    15: "찬타",
    16: "일기통관",
    17: "삼색동순",
    18: "더블리치",
    19: "삼색동각",
    20: "산깡즈",
    21: "또이또이",
    22: "산안커",
    23: "소삼원",
    24: "혼노두",
    25: "치또이츠",
    26: "준찬타",
    27: "혼일색",
    28: "량페코",
    29: "청일색",
    30: "일발",
    31: "도라",
    32: "적도라",
    33: "뒷도라",
    34: "북도라",
    35: "천화",
    36: "지화",
    37: "대삼원",
    38: "사암각",
    39: "자일색",
    40: "녹일색",
    41: "청노두",
    42: "국사무쌍",
    43: "소사희",
    44: "사공자",
    45: "구련보등",
    46: "팔연장",
    47: "순정 구련보등",
    48: "사암각단기",
    49: "국사무쌍 13면",
    50: "대사희",
    51: "츠바메가에시",
    52: "영상개론",
    53: "십이낙태",
    54: "오문제",
    55: "삼련각",
    56: "일색삼순",
    57: "일통모월",
    58: "구통로어",
    59: "인화",
    60: "대차륜",
    61: "대죽림",
    62: "대수린",
    63: "돌 위에서 삼년",
    64: "대칠성",
}

IMPORTANT_YAKUS = [
    "천화",
    "지화",
    "대삼원",
    "사암각",
    "자일색",
    "녹일색",
    "청노두",
    "국사무쌍",
    "소사희",
    "사공자",
    "구련보등",
    "팔연장",
    "순정 구련보등",
    "사암각단기",
    "국사무쌍 13면",
    "대사희",
    "츠바메가에시",
    "영상개론",
    "십이낙태",
    "오문제",
    "삼련각",
    "일색삼순",
    "일통모월",
    "구통로어",
    "인화",
    "대차륜",
    "대죽림",
    "대수린",
    "돌 위에서 삼년",
    "대칠성",
]

# ref: https://github.com/yf-dev/majsoul-record/blob/master/src/consts.py
